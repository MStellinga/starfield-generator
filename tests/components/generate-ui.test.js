/* eslint-disable jest/valid-expect */
import { h } from 'preact';
import render from 'preact-render-to-string';
import GeneratorUI from '../../src/components/generator-ui';

describe('Test Generator UI', () => {
	test('Generator UI renders', () => {
        const tree = render(<GeneratorUI />);
        expect(tree).toMatchSnapshot();	
    });

    // Unfortunately, adding tests that call event handlers is hard to do with Enzyme and function-based Preact components (at this time, anyway)
});
