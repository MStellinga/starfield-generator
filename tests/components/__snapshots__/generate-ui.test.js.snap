// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Generator UI Generator UI renders 1`] = `"<div><div class=\\"container\\"><div class=\\"section\\"><table><tbody><tr><td>Bubble sizes</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"30\\" /></td><td>-</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"160\\" /></td><td colspan=\\"3\\">Fractal division count</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"6\\" /></td></tr><tr><td>Dark center size</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"30\\" /></td><td>-</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"80\\" /></td><td colspan=\\"3\\">Fractal min size</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"5\\" /></td></tr><tr><td colspan=\\"3\\">Bubble distance fall-off</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"2\\" /></td><td colspan=\\"3\\">Fractal color gain (%)</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"10\\" /></td></tr><tr><td colspan=\\"3\\">Bubble color desaturate</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"4\\" /></td><td colspan=\\"4\\"></td></tr><tr><td colspan=\\"4\\"><div class=\\"checkbox\\"><input type=\\"checkbox\\" /><span>Blur nebula</span></div></td><td colspan=\\"4\\"><div class=\\"checkbox\\"><input type=\\"checkbox\\" /><span>Blur fractal</span></div></td></tr></tbody></table></div><div class=\\"section\\"><h2>Clusters</h2><table class=\\"clusters\\"><thead><tr><th></th><th>Power</th><th>Stars 1</th><th>Stars 2</th><th>Stars 3</th><th>Bubbles</th><th>Fractal</th><th>Color</th><th></th></tr></thead><tbody><tr><td></td><td></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"2000\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"100\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"50\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><div><div class=\\"pickerButton\\" style=\\"background-color: rgb(255,255,255);\\"></div></div></td><td></td></tr><tr><td>1</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"100\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"300\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"50\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"10\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"500\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><div><div class=\\"pickerButton\\" style=\\"background-color: rgb(0,200,250);\\"></div></div></td><td><button>X</button></td></tr><tr><td>2</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"30\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"300\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"50\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"10\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"500\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><div><div class=\\"pickerButton\\" style=\\"background-color: rgb(0,150,200);\\"></div></div></td><td><button>X</button></td></tr><tr><td>3</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"300\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"300\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"50\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"10\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"500\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><div><div class=\\"pickerButton\\" style=\\"background-color: rgb(0,0,250);\\"></div></div></td><td><button>X</button></td></tr><tr><td>4</td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"300\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"0\\" /></td><td><input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"100\\" /></td><td><div><div class=\\"pickerButton\\" style=\\"background-color: rgb(230,0,40);\\"></div></div></td><td><button>X</button></td></tr></tbody></table></div><div class=\\"section\\"><div>Image size <input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"800\\" /> x <input type=\\"text\\" pattern=\\"-?[0-9\\\\.]{1,5}\\" class=\\"smallfield\\" value=\\"400\\" /></div></div><div class=\\"section\\"><label>Generate</label><button id=\\"button-generate-stars\\">Stars</button><button id=\\"button-generate-nebula\\">Bubble nebula</button><button id=\\"button-generate-fractal\\">Fractal nebula</button></div><div class=\\"progress\\"><div class=\\"progressdone\\" style=\\"width: 0%;\\"></div></div></div><div class=\\"picture\\" style=\\"width: 800px;\\"><div class=\\"pictureEditButtons\\"><button>Add</button><button>Move</button></div><canvas width=\\"800\\" height=\\"400\\" class=\\"starcanvas\\"></canvas><div class=\\"pictureEditButtons\\"><a href download=\\"Starfield.png\\"><button>Save to File</button></a></div></div></div>"`;
